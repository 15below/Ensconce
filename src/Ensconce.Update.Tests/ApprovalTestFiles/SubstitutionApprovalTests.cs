using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.TestFrameworks;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;
using System.Xml.Linq;

namespace Ensconce.Update.Tests.ApprovalTestFiles
{
    [TestFixture]
    [Category("Approval Test")]
    public class SubstitutionApprovalTests
    {
        [SetUp]
        public void Setup()
        {
            Environment.CurrentDirectory = TestContext.CurrentContext.TestDirectory;
        }

        [Test]
        [UseReporter(typeof(NUnitReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Transform()
        {
            var config = TagDictionary.FromSources("ident", new Dictionary<TagSource, string> { { TagSource.XmlFileName, @"ApprovalTestFiles\Config.xml" } });
            var lazyConfig = new Lazy<TagDictionary>(() => config);
            var result = XDocument.Parse(ProcessSubstitution.Update(@"ApprovalTestFiles\Substitution.xml", @"ApprovalTestFiles\BaseFile.xml", lazyConfig));
            Approvals.VerifyXml(result.ToString());
        }
    }
}
