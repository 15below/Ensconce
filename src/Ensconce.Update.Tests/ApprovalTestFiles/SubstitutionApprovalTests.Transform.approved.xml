<root>
  <node value="XX-LOC" prop="MyValue" instanceProp="MyInstance1Value" />
  <node2 value1="LOC" value2="NotSet"></node2>
</root>